src/main/java/com/example/bookstore/controller/BookController.java

```diff
@@ -24,35 +25,40 @@
 @Tag(name = "Book management", description = "Endpoints for managing products")
 @RequiredArgsConstructor
 @RestController
-@RequestMapping(value = "/books")
+@RequestMapping(value = "/api/books")
 public class BookController {
     private final BookService bookService;
 
+    @PreAuthorize("hasRole('ROLE_USER')")
     @Operation(summary = "Get books", description = "Get a list of all available books")
     @GetMapping
     public List<BookDto> getAll(Pageable pageable) {
         return bookService.getAll(pageable);
     }
 
+    @PreAuthorize("hasRole('USER')")
     @Operation(summary = "Get book", description = "get book by id")
     @GetMapping("/{id}")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
+    @PreAuthorize("hasRole('ADMIN')")
```

Comment:
```suggestion
    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
```

Reply:
I doubt it's necessary to use it in create, but I get it